nfs/src/rpc_task.cpp

```diff
@@ -517,7 +626,8 @@ static void lookup_callback(
      * Kernel must cache -ve entries.
      */
     const bool cache_negative =
-        (aznfsc_cfg.lookupcache_int == AZNFSCFG_LOOKUPCACHE_ALL);
+        ((aznfsc_cfg.lookupcache_int == AZNFSCFG_LOOKUPCACHE_ALL) &&
+        (task->get_proxy_op_type() == FUSE_LOOKUP));
```

Comment:
we should check for get_op_type(). I know currently you have proxy type same as optype for non-proxy requests, but still the check is logically wrong.